lib/eosknowledgeprivate/ekn-web-view-dialog.c

```diff
@@ -0,0 +1,256 @@
+                                   PROP_WEBVIEW,
+                                   g_param_spec_object ("webview",
+                                                        "Web View",
+                                                        "The WebkitWebView internal child",
+                                                        WEBKIT_TYPE_WEB_VIEW,
+                                                        G_PARAM_READABLE));
+  g_object_class_install_property (object_class,
+                                   PROP_HOME_URI,
+                                   g_param_spec_string ("home-uri",
+                                                        "Home uri",
+                                                        "The home uri to display",
+                                                        NULL,
+                                                        G_PARAM_READWRITE));
+
+  gtk_widget_class_set_template_from_resource (widget_class, "/com/endlessm/knowledge-private/EknWebViewDialog.ui");
+
+  gtk_widget_class_bind_template_child_private (widget_class, EknWebViewDialog, webview);
+  gtk_widget_class_bind_template_child_private (widget_class, EknWebViewDialog, headerbar);
+  gtk_widget_class_bind_template_child_private (widget_class, EknWebViewDialog, back);
+  gtk_widget_class_bind_template_child_private (widget_class, EknWebViewDialog, forward);
+
+  gtk_widget_class_bind_template_callback (widget_class, on_back_clicked);
+  gtk_widget_class_bind_template_callback (widget_class, on_forward_clicked);
+  gtk_widget_class_bind_template_callback (widget_class, on_home_clicked);
+}
+
+/********************************* Public API *********************************/
+
+/**
+ * ekn_web_view_dialog_get_webview:
+ * @dialog: a #EknWebViewDialog
+ *
+ * Returns the Webkit web view used to display the URI
+ *
+ * Returns: (transfer none): the webview is owned by the dialog.
+ */
+WebKitWebView *
+ekn_web_view_dialog_get_webview (EknWebViewDialog *dialog)
+{
+  g_return_val_if_fail (EKN_IS_WEB_VIEW_DIALOG (dialog), NULL);
+  return dialog->priv->webview;
+}
+
+/**
+ * ekn_web_view_dialog_get_home_uri:
+ * @dialog: a #EknWebViewDialog
+ *
+ * Returns the home uri displayed in the web view.
+ *
+ */
+const gchar *
+ekn_web_view_dialog_get_home_uri (EknWebViewDialog *dialog)
+{
+  g_return_val_if_fail (EKN_IS_WEB_VIEW_DIALOG (dialog), NULL);
+  return dialog->priv->home_uri;
+}
+
+/**
+ * ekn_web_view_dialog_set_home_uri:
+ * @dialog: a #EknWebViewDialog
+ * @uri:
+ *
+ * Sets the home uri to display in the web view.
+ *
+ */
+void
+ekn_web_view_dialog_set_home_uri (EknWebViewDialog *dialog, const gchar *uri)
+{
+  EknWebViewDialogPrivate *priv;
+
+  g_return_if_fail (EKN_IS_WEB_VIEW_DIALOG (dialog));
+
+  priv = dialog->priv;
+
+  if (g_strcmp0 (priv->home_uri, uri))
+    {
+      g_free (priv->home_uri);
+      g_free (priv->home_uri_scheme);
+
+      priv->home_uri = g_strdup (uri);
+      priv->home_uri_scheme = g_uri_parse_scheme (uri);
```

Comment:
ohh good catch, thats some leftover from when I called gtk_show_uri() from this object.
Then I decided that behavior was to specific to be in the widget itself and moved that logic to JS